internal/testutil/acc/cluster.go

```diff
@@ -136,44 +136,42 @@ func (r *ReplicationSpecRequest) AddDefaults() {
 	}
 }
 
-func (r *ReplicationSpecRequest) AllRegionConfigs() []admin.CloudRegionConfig {
+func (r *ReplicationSpecRequest) AllRegionConfigs() []admin.CloudRegionConfig20240805 {
 	config := cloudRegionConfig(*r)
-	configs := []admin.CloudRegionConfig{config}
+	configs := []admin.CloudRegionConfig20240805{config}
 	for i := range r.ExtraRegionConfigs {
 		extra := r.ExtraRegionConfigs[i]
 		configs = append(configs, cloudRegionConfig(extra))
 	}
 	return configs
 }
 
-func replicationSpec(req *ReplicationSpecRequest) admin.ReplicationSpec {
+func replicationSpec(req *ReplicationSpecRequest) admin.ReplicationSpec20240805 {
 	if req == nil {
 		req = new(ReplicationSpecRequest)
 	}
 	req.AddDefaults()
-	defaultNumShards := 1
 	regionConfigs := req.AllRegionConfigs()
-	return admin.ReplicationSpec{
-		NumShards:     &defaultNumShards,
+	return admin.ReplicationSpec20240805{
```

Comment:
Removed NumShards as new SDK ReplicationSpec model no longer defines it.
The ClusterResourceHcl was generating all config with num_shards = 1, since this is equivalent to not defining the attribute no risk in removing.